sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -769,12 +675,68 @@ int gcoap_req_init(coap_pkt_t *pdu, uint8_t *buf, size_t len,
 
 ssize_t gcoap_finish(coap_pkt_t *pdu, size_t payload_len, unsigned format)
 {
-    /* reconstruct full PDU buffer length */
-    size_t len = pdu->payload_len + (pdu->payload - (uint8_t *)pdu->hdr);
+    unsigned format_offset = UINT_MAX;
+    unsigned len_delta = 0;     /* buffer length to move forward */
+    uint8_t *move_pos = 0;      /* buffer position to move forward */
+
+    /* Remove or adjust Content-Format option for request, if any.
+     * Assumes current payload format is COAP_FORMAT_NO_PAYLOAD */
```

Comment:
I had a hard time understanding the code below. Maybe some additional comments would help the next person. Something like:

```
/* Remove or adjust Content-Format option for request, if any.
 * Assumes current payload format is COAP_FORMAT_NO_PAYLOAD
 * First the content format is updated and the option size difference is
 * stored in len_delta. Then the Option array is updated. Finally the options as 
 * placed in the buffer are moved to their updated position (offset) */
```

Feel free to adjust to your liking

Reply:
Yes, as I described in a reply above.